plugins/techdocs/src/reader/components/TechDocsSearch.test.tsx

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import { TechDocsSearch } from './TechDocsSearch';
+import {
+  act,
+  fireEvent,
+  render,
+  waitFor,
+  within,
+} from '@testing-library/react';
+import { wrapInTestApp } from '@backstage/test-utils';
+import { ApiProvider, ApiRegistry } from '@backstage/core-app-api';
+import { searchApiRef } from '@backstage/plugin-search';
+
+const entityId = {
+  name: 'test',
+  namespace: 'testspace',
+  kind: 'Testable',
+};
+
+const emptyResults = Promise.resolve({
+  results: [],
+});
+const singleResult = Promise.resolve({
+  results: [
+    {
+      type: 'testsearchresult',
+      document: {
+        text: 'text',
+        title: 'title',
+        location: '/something/something',
+      },
+    },
+  ],
+});
+
+describe('<TechDocsPage />', () => {
```

Comment:
I think these are tests specifically for TechDocsSearch and not the TechDocsPage, it makes sense though to write some tests around the `context`, as it should navigate to different urls depending on that, wanna add that?

Reply:
Yup, makes sense. The material-ui component used for this makes this complex to go via user actions but testing the logic separately is worth it so I'll pull that into a separate fn.